src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -334,30 +340,74 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
     }
 
     @Override
-    public Map<String, Long> materializePositions(final String topicId, final SubscriptionBase.InitialPosition position)
+    public Map<String, Long> materializePositions(final EventType eventType, final Subscription subscription)
```

Comment:
No, it doesn't belong, but I see no problems to use Subscription on repository layer. Subscription belongs to domain layer and I think we can use it in repositories.